src/storage/txn/actions/flashback_to_version.rs

```diff
@@ -291,9 +306,32 @@ pub mod tests {
         // Since the key has been deleted, flashback to version 1 should not do
         // anything.
         assert_eq!(
-            must_flashback_write(&engine, k, ts, *ts.incr(), *ts.incr()),
+            must_flashback_to_version(&engine, k, ts, *ts.incr(), *ts.incr()),
             0
         );
         must_get_none(&engine, k, ts);
     }
+
+    #[test]
+    fn test_flashback_to_version_pessimistic() {
+        use kvproto::kvrpcpb::PrewriteRequestPessimisticAction::*;
+
+        let engine = TestEngineBuilder::new().build().unwrap();
+        let (k, v) = (b"k", b"v");
+        // Prewrite and commit Put(k -> v1) with stat_ts = 10, commit_ts = 20.
+        must_prewrite_put(&engine, k, v, k, 10);
+        must_commit(&engine, k, 10, 20);
+
+        let v1 = b"v1";
+        must_acquire_pessimistic_lock(&engine, k, k, 30, 30);
+        must_pessimistic_locked(&engine, k, 30, 30);
+
+        // Flashback to version 25 with start_ts = 30, commit_ts = 40.
+        assert_eq!(must_flashback_to_version(&engine, k, 25, 30, 40), 1);
+
+        // Pessimistic Prewrite Put(k -> v1) with stat_ts = 30 will be error with
+        // Rollback.
+        must_pessimistic_prewrite_put_err(&engine, k, v1, k, 30, 30, DoPessimisticCheck);
+        must_get(&engine, k, 45, v);
```

Comment:
Better write two MVCC values before the pessimistic lock test and flashback to the first one, so that we can test whether the flashback works well.

Reply:
Done.